### PR TITLE
Answers to Prob Set 1

### PR DESCRIPTION
**Pull request by chrismit258**

### CODE REVIEW

File: answers.sh [hunk 1]
```diff
@@ -1,17 +1,81 @@
...
+states="$datasets/misc/states.tab.gz"
 
-answer_1=$(gzcat $states \
-    | grep -v '^#' \
-    | cut -f1,2 \
-    | sort -k2n \
-    | tail -n1 \
-    | cut -f1 \
-    | sed 's/"//g') 
+answer_1=$(zcat $states | cut -f1,6 | sort -k2n | grep -v '^#' | head -n1 | cut -f1 | sed 's/"//g' | sed 's/_/ /g')
```
Comment (kriemo) [inline answers.sh]: correct, but it is a good habit to split the code across multiple lines for readability. 

```bash
zcat $states \
  | cut -f1,6 \
  | sort -k2n \
  | grep -v '^#' \
  | head -n1 \
  | cut -f1 \
  | sed 's/"//g' \
  | sed 's/_/ /g'
```

File: answers.sh [hunk 2]
```diff
@@ -1,17 +1,81 @@
...
 
 echo "answer-1: $answer_1"
 
+# How many sequence records are in the sample.fa file? 
+samples="$datasets/fasta/sample.fa"
+
+answer_2=$(cat $samples | grep "^>" | wc -l)
```
Comment (kriemo) [inline answers.sh]: good. When pipeing uncompressed data, I would avoid using cat and instead list the filename after `grep`. It's not wrong just unnecessary.

```bash
grep "^>" $samples | wc -l
```

File: answers.sh [hunk 3]
```diff
@@ -1,17 +1,81 @@
...
+
+echo "answer-2: $answer_2"
+
+# How many unique CpG IDs are in cpg.bed.gz? 
+cpg="$datasets/bed/cpg.bed.gz"
+
+answer_3=$(zcat $cpg | cut -f4 | sort -g | uniq | wc -l)
```
Comment (kriemo) [inline answers.sh]: also correct, but a minor point. Use `sort -g` when sorting numbers with decimals and scientific notation, in this case the text is alphanumeric so no need to sort by `-g`.

File: answers.sh [hunk 4]
```diff
@@ -1,17 +1,81 @@
...
+
+answer_4=$(cat $SP1 | grep '^@cluster' | wc -l)
+
+echo "answer-4: $answer_4"
+
+# How many words are on lines containing the word bloody in hamlet.txt? 
+
+hamlet="$datasets/misc/hamlet.txt"
+
+answer_5=$(cat $hamlet | grep 'bloody' | wc -w)
```
Comment (kriemo) [inline answers.sh]: this question is a little tricky. use a case-insensitive search to find all instances of `bloody`. see `man grep`

File: answers.sh [hunk 5]
```diff
@@ -1,17 +1,81 @@
...
+
+echo "answer-5: $answer_5"
+
+# What is the length of the sequence in the first record of sample.fa? 
+
+answer_6=$(head -n2 $samples | grep -v '^>' | wc -c)
```
Comment (kriemo) [inline answers.sh]: there is a hidden newline `\n` character that is being counted. use `tr` to get rid of it before counting

Comment (chrismit258) [inline answers.sh]: I don't understand this issue. I added tr -d [\n] before the wc command, but the character count didn't change.

Comment (kriemo) [inline answers.sh]: try the `tr` command with `tr -d '\n'`

**Comment by jayhesselberth** [comment]
FYI `wc -m` just counts characters, no need for `tr` in that case.

**Comment by kriemo** [comment]
Strong work 👍

**Pull request closed without merging**